### PR TITLE
Ibp 3951 plot entry replacement

### PR DESCRIPTION
**Pull request by clarysabel**

### CODE REVIEW

File: src/main/java/com/efficio/fieldbook/web/common/controller/ExportStudyGermplasmController.java [hunk 1]
```diff
@@ -103,7 +107,7 @@ protected String doExport(final int exportType, final HttpServletResponse respon
 
 		final int studyId = this.userSelection.getWorkbook().getStudyDetails().getId();
 		final boolean hasStocks =
```
Comment (darla-leafnode) [inline src/main/java/com/efficio/fieldbook/web/common/controller/ExportStudyGermplasmController.java:109]: While we're standardizing nomenclature, let's rename:
1. `ExportStudyGermplasmController` -> `ExportStudyEntriesController`
2. `hasStocks` variable -> `studyHasEntries`

File: src/main/java/com/efficio/fieldbook/web/common/service/impl/CrossingTemplateExcelExporter.java [hunk 2]
```diff
@@ -302,7 +306,7 @@ private FileExportInfo createExcelOutputFile(final String studyName, final Workb
 	}
 
 	void validateIfStudyHasGermplasm(final Integer studyId) throws CrossingTemplateExportException {
```
Comment (darla-leafnode) [inline src/main/java/com/efficio/fieldbook/web/common/service/impl/CrossingTemplateExcelExporter.java:308]: Rename to `validateIfStudyHasEntries`

File: src/main/java/com/efficio/fieldbook/web/trial/controller/ImportGermplasmListController.java [hunk 3]
```diff
@@ -274,11 +284,11 @@ private void saveStudyGermplasm(final int studyId) {
 			final List<ImportedGermplasm> importedGermplasm = importedGermplasmList.getImportedGermplasms();
 			final List<StudyGermplasmDto> studyGermplasmDtoList = this.studyGermplasmTransformer.transformToStudyGermplasmDto(importedGermplasm);
 			// Delete the existing stocks so that we can replace it with the current list.
-			this.studyGermplasmService.deleteStudyGermplasm(studyId);
```
Comment (darla-leafnode) [inline src/main/java/com/efficio/fieldbook/web/trial/controller/ImportGermplasmListController.java:277]: As long as we are standardizing, can we rename StudyGermplasmService and StudyGermplasmServiceImpl (and its variable declaration in using classes) to `StudyEntriesService` and `StudyEntriesServiceImpl`

File: src/main/resources/Messages.properties [hunk 4]
```diff
@@ -1570,6 +1572,7 @@ environment.has.plot.data=All observations and/or fieldmap for this environment
 study.should.have.at.least.one.environment=Study should have at least one environment.
 
 crosses.and.selections.no.germplasm.error=Please select at least one germplasm.
+study.observations.changing.plot.entry.warning=If you change the entry all the information related to these plots (including observations) will be now referenced to the new entry. Are you sure you want to proceed?
```
Comment (darla-leafnode) [inline src/main/resources/Messages.properties:1575]: Improve grammar:
If you change the entry**,** all the information related to these plots (including observations) will **now refer to** the new entry. Are you sure you want to proceed?

File: src/test/java/com/efficio/fieldbook/web/common/controller/ExportStudyGermplasmControllerTest.java [hunk 5]
```diff
@@ -34,14 +33,23 @@
...
 import java.io.IOException;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.Map.Entry;
+import java.util.Random;
 
 public class ExportStudyGermplasmControllerTest {
```
Comment (darla-leafnode) [inline src/test/java/com/efficio/fieldbook/web/common/controller/ExportStudyGermplasmControllerTest.java:54]: Rename as well to `ExportStudyEntriesControllerTest`

**Review by darla-leafnode — COMMENTED** [review]
Just some minor comments here @clarysabel - please check

**Comment by clarysabel** [comment]
Hi @darla-leafnode Thanks for reviewing this PR, we did not rename the classes at this instance because it affected many other classes and we are pretty much on time with this ticket. I will open a new ticket for that

**Pull request merged**